aggregator/src/aggregator/aggregation_job_creator.rs

```diff
@@ -3078,29 +3538,46 @@ mod tests {
         )
         .unwrap();
 
-        // Verify that all reports we saw a report aggregation for are scrubbed.
+        // Verify that all reports we saw a report aggregation for are scrubbed, if the aggregation
+        // parameter is not the unit type.
         let all_seen_report_ids: HashSet<_> = agg_jobs_and_report_ids
             .iter()
-            .flat_map(|(_, report_aggs)| report_aggs.iter().map(|ra| ra.report_id()))
+            .flat_map(|(agg_job, report_aggs)| {
+                report_aggs.iter().map(|ra| {
+                    (
+                        (agg_job.aggregation_parameter() as &A::AggregationParam).clone(),
+                        ra.report_id(),
+                    )
+                })
+            })
             .collect();
-        for report_id in &all_seen_report_ids {
-            tx.verify_client_report_scrubbed(task_id, report_id).await;
+        for (_, report_id) in &all_seen_report_ids {
+            if is_unit_type::<A::AggregationParam>() {
```

Comment:
nit: hoist these aggregation-param-is-unit-type checks outside of the loops they are contained in, since these are effectively constant values.